libraries/core_libs/consensus/src/dag/block_proposer.cpp

```diff
@@ -107,6 +108,12 @@ bool SortitionPropose::propose() {
     return false;
   }
   LOG(log_nf_) << "VDF computation time " << vdf.getComputationTime() << " difficulty " << vdf.getDifficulty();
+  auto pk = key_manager_->get(*proposal_period, node_addr_);
+  if (pk && *pk != vrf_pk_) {
```

Comment:
why it is here and not at the beginning of the function ? Some processing can be saved

Reply:
inst it even better when the message is printed all the time ? It happens when user somehow adjust his wallet in wrong way or he registers his validator in wrong way, in which case it is good to get his attention as soon as possible so he can change it

Reply:
ok, moved it up